src/Maps/components/MapsContainer.ts

```diff
@@ -0,0 +1,242 @@
+
+import { LeafletMap } from "./LeafletMap";
+import googleApiWrapper from "./GoogleMap";
+import { Container } from "../utils/namespace";
+import { fetchData, fetchMarkerObjectUrl, parseStaticLocations } from "../utils/Data";
+import { validLocations, validateLocationProps } from "../utils/Validations";
+import MapsContainerProps = Container.MapsContainerProps;
+import MapProps = Container.MapProps;
+import Location = Container.Location;
+import DataSourceLocationProps = Container.DataSourceLocationProps;
+
+// tslint:disable-next-line:no-submodule-imports
+import "leaflet/dist/leaflet.css";
+// Re-uses images from ~leaflet package
+// Use workaround for marker icon, that is not standard compatible with webpack
+// https://github.com/ghybs/leaflet-defaulticon-compatibility#readme
+import "leaflet-defaulticon-compatibility";
+// tslint:disable-next-line:no-submodule-imports
+import "leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.webpack.css";
+import "../../ui/Maps.css";
+
+export interface MapsContainerState {
+    alertMessage?: string;
+    locations: Location[];
+    isFetchingData?: boolean;
+}
+
+export default class MapsContainer extends Component<MapsContainerProps, MapsContainerState> {
+    private subscriptionHandles: number[] = [];
+    readonly state: MapsContainerState = {
+        alertMessage: "",
+        locations: [],
+        isFetchingData: false
+    };
+
+    render() {
+        const mapsApiToken = this.props.apiToken ? this.props.apiToken.replace(/ /g, "") : undefined;
+        const commonProps = {
+            allLocations: this.state.locations,
+            fetchingData: this.state.isFetchingData,
+            className: this.props.class,
+            alertMessage: this.state.alertMessage,
+            divStyles: parseStyle(this.props.style),
+            onClickMarker: this.onClickMarker,
+            mapsToken: mapsApiToken,
+            ...this.props as MapProps
+        };
+
+        return this.props.mapProvider === "googleMaps"
+            ? createElement(googleApiWrapper, { ...commonProps })
+            : createElement(LeafletMap, { ...commonProps });
+    }
+
+    componentWillReceiveProps(nextProps: MapsContainerProps) {
+        this.resetSubscriptions(nextProps.mxObject);
+        const validationMessage = validateLocationProps(nextProps);
+        if (nextProps && nextProps.mxObject) {
+            if (validationMessage) {
+                this.setState({ alertMessage: validationMessage });
+            } else {
+                this.fetchData(nextProps.mxObject);
+            }
+        } else {
+            this.setState({ locations: [], alertMessage: "", isFetchingData: false });
+        }
+    }
+
+    componentWillUnmount() {
+        this.unsubscribeAll();
+    }
+
+    private resetSubscriptions(contextObject?: mendix.lib.MxObject) {
+        this.unsubscribeAll();
+        if (this.props.locations && this.props.locations.length) {
+            if (contextObject) {
```

Comment:
Any reason u don't have this check on the previous line? There're no else paths, so move this to outer check

Reply:
done